chrome/content/zotero/xpcom/prefs.js

```diff
@@ -542,4 +544,50 @@ Zotero.Prefs = new function() {
 			return `{{ ${[field, truncate, prefix, suffix].filter(f => f !== null).join(' ')} }}`;
 		});
 	};
+
+	/**
+	 * Preferences set in <profile>/user.js override prefs.js, so they can't
+	 * be changed via the UI and may be hard to locate and remove. Warn if
+	 * user.js contains user_pref() directives.
+	 */
+	this._checkUserJS = async function () {
+		let userJSPath = PathUtils.join(Zotero.Profile.dir, 'user.js');
+		let userJS;
+		try {
+			userJS = await Zotero.File.getContentsAsync(userJSPath);
+		}
+		catch {
+			return;
+		}
+		
+		const PREF_RE = /^\s*user_pref\s*\(\s*['"]extensions\.zotero\..+$/g;
+		
+		let updatedUserJS = userJS
+			.split('\n')
+			.filter((prefLine) => {
+				if (PREF_RE.test(prefLine)
+						&& prefLine.includes('extensions.zotero.fileHandler.')
+						&& prefLine.toLowerCase().includes('scholaread')) {
+					Zotero.debug('user.js contains disallowed pref: ' + prefLine);
+					return false;
+				}
+				return true;
+			})
+			.join('\n');
+		
+		if (updatedUserJS === userJS) {
+			return;
+		}
+		
+		await Zotero.File.putContentsAsync(userJSPath, updatedUserJS);
```

Comment:
This should probably be in a try/catch as well. Since this more or less by definition will involve the file being written by another program, it's not impossible it could become unwritable, and we don't want to break startup while trying to fix it.